Python/CosmosDB-NoSQL_VectorSearch/CosmosDB-NoSQL-Quickstart-RAG-Chatbot.ipynb

```diff
@@ -0,0 +1,619 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import the required libraries\n",
+    "import time\n",
+    "import json\n",
+    "import uuid\n",
+    "import urllib \n",
+    "import ijson\n",
+    "import zipfile\n",
+    "from dotenv import dotenv_values\n",
+    "from openai import AzureOpenAI\n",
+    "from azure.core.exceptions import AzureError\n",
+    "from azure.cosmos import PartitionKey, exceptions\n",
+    "from time import sleep\n",
+    "import gradio as gr\n",
+    "\n",
+    "# Cosmos DB imports\n",
+    "from azure.cosmos import CosmosClient\n",
+    "\n",
+    "# Load configuration\n",
+    "env_name = \"sample_env_file.env\"\n",
+    "config = dotenv_values(env_name)\n",
+    "\n",
+    "cosmos_conn = config['cosmos_uri']\n",
+    "cosmos_key = config['cosmos_key']\n",
+    "cosmos_database = config['cosmos_database_name']\n",
+    "cosmos_collection = config['cosmos_collection_name']\n",
+    "cosmos_vector_property = config['cosmos_vector_property_name']\n",
+    "comsos_cache_db = config['cosmos_cache_database_name']\n",
+    "cosmos_cache = config['cosmos_cache_collection_name']\n",
+    "\n",
+    "# Create the Azure Cosmos DB for NoSQL async client for faster data loading\n",
+    "cosmos_client = CosmosClient(url=cosmos_conn, credential=cosmos_key)\n",
+    "\n",
+    "openai_endpoint = config['openai_endpoint']\n",
+    "openai_key = config['openai_key']\n",
+    "openai_api_version = config['openai_api_version']\n",
+    "openai_embeddings_deployment = config['openai_embeddings_deployment']\n",
+    "openai_embeddings_dimensions = int(config['openai_embeddings_dimensions'])\n",
+    "openai_completions_deployment = config['openai_completions_deployment']\n",
+    "\n",
+    "# Movies file url\n",
+    "storage_file_url = config['storage_file_url']\n",
+    "\n",
+    "# Create the OpenAI client\n",
+    "openai_client = AzureOpenAI(azure_endpoint=openai_endpoint, api_key=openai_key, api_version=openai_api_version)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "011ec9d5-8fe7-406f-85a7-027b84a5189f",
+   "metadata": {},
+   "source": [
+    "#  Create a database and containers with vector policies\n",
+    "\n",
+    "This function takes a database object, a collection name, the name of the document property that will store vectors, and the number of vector dimensions used for the embeddings."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "305807f8-0205-481a-9d71-e8b88b504019",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "db = cosmos_client.create_database_if_not_exists(cosmos_database)\n",
+    "\n",
+    "# Create the vector embedding policy to specify vector details\n",
+    "vector_embedding_policy = {\n",
+    "    \"vectorEmbeddings\": [ \n",
+    "        { \n",
+    "            \"path\":\"/\" + cosmos_vector_property,\n",
+    "            \"dataType\":\"float32\",\n",
+    "            \"distanceFunction\":\"dotproduct\",\n",
```

Comment:
Is `dotproduct` important to this? I ask because there is a bug where the distance function specified here is ignored in some cases right now.

Reply:
No, we could use a different one. Will discuss with James.